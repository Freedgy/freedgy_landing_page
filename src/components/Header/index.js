import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import LanguageSelector from '../LanguageSelector';

const lightColor = 'rgba(255, 255, 255, 1)';

const styles = theme => ({
    link: {
        color: lightColor,
        '&:hover': {
            color: theme.palette.common.white,
        },
    },
    button: {
        borderColor: lightColor,
    },
    logo: {
        height: '150px',
        width: '150px',
        img: {
            margin: 0,
            height: '455px',
            width: 100
        },
    },
    appBarTransparent: {
        backgroundColor: 'transparent',
        boxShadow: 'none',
        color: 'white'
    },
    appBarSolid: {
        backgroundColor: 'white',
        color: 'black'
    }
});

function Header({ classes, title, lang, location = null }) {
    const [navBackground, setNavBackground] = useState('appBarTransparent')
    const navRef = React.useRef()
    navRef.current = navBackground
    useEffect(() => {
        const handleScroll = () => {
            const show = window.scrollY > 100
            if (show) {
                setNavBackground('appBarSolid')
            } else {
                setNavBackground('appBarTransparent')
            }
        }
        document.addEventListener('scroll', handleScroll)
        return () => {
            document.removeEventListener('scroll', handleScroll)
        }
    }, [])

    return (
        <AppBar position="sticky" className={classes[navRef.current]} elevation={0}>
            <Toolbar>
                <Grid container spacing={8} alignItems="center">
                    <Grid item>
                        <Avatar className={classes.logo} src="/images/logo_base.png" />
                    </Grid>
                    <Grid>
                        <IconButton color="inherit">
                            1
                        </IconButton>
                        <IconButton color="inherit">
                            2
                        </IconButton>
                        <IconButton color="inherit">
                            3
                        </IconButton>
                        <IconButton color="inherit">
                            4
                        </IconButton>
                        <IconButton color="inherit">
                            5
                        </IconButton>
                        <IconButton color="inherit">
                            6
                        </IconButton>
                    </Grid>
                    <Grid item>
                        <LanguageSelector classes={classes} className={classes.link} location={location} lang={lang} />
                    </Grid>
                    <Grid item>
                        <Button className={classes.button} variant="outlined" color="inherit" size="small">
                            run
                        </Button>
                    </Grid>
                </Grid>
            </Toolbar>
        </AppBar>
    );
}

Header.propTypes = {
    classes: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
};

export default withStyles(styles)(Header);
