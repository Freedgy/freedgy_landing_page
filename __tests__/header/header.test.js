import React from "react"
import renderer from "react-test-renderer"

import Header from "../../src/components/Header"

import jsdom from 'jsdom';
const {JSDOM} = jsdom;
const {document} = (new JSDOM('<!doctype html><html><body></body></html>')).window;
global.document = document;
global.window = document.defaultView;
global.Image = window.Image;

describe("Header", () => {
    it("renders correctly", async () => {
        const datas = require("../../static/pages/pages.en.json");

        const tree = renderer
            .create(<Header lang="en" datas={datas.layout.header} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})