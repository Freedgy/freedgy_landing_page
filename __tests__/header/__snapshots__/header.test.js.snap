// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary Header-appBarTransparent-2 mui-fixed MuiPaper-elevation0"
>
  <div
    className="MuiBox-root MuiBox-root-8 Header-container-4"
  >
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <img
        alt="watermark_freedgy"
        className="Header-logo-1"
        src="/images/logo/watermark.png"
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-10 Header-navBar-5"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-6"
        href="#home"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          home
        </span>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-6"
        href="#about"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          about
        </span>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-6"
        href="#team"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          team
        </span>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-6"
        href="#contact"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          contact
        </span>
      </a>
    </div>
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <div
        className="ReactFlagsSelect-module_flagsSelect__2pfa2"
        data-testid="rfs"
      >
        <button
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-labelledby="rfs-btn"
          className="ReactFlagsSelect-module_selectBtn__19wW7"
          data-testid="rfs-btn"
          disabled={false}
          id="rfs-btn"
          onClick={[Function]}
          onKeyUp={[Function]}
          style={
            Object {
              "fontSize": 16,
            }
          }
          type="button"
        >
          <span
            className="ReactFlagsSelect-module_selectValue__152eS"
          >
            Select a country
          </span>
        </button>
      </div>
    </div>
  </div>
</header>
`;
